src/com/google/javascript/jscomp/SourceFile.java

```diff
@@ -378,7 +379,7 @@ private static SourceFile fromZipEntry(String zipURL, Charset inputCharset) {
   public static SourceFile fromZipEntry(
       String originalZipPath, String absoluteZipPath, String entryPath, Charset inputCharset)
       throws MalformedURLException {
-    String zipEntryPath = JAR_URL_PREFIX + absoluteZipPath + BANG_SLASH + entryPath;
+    String zipEntryPath = JAR_URL_PREFIX + absoluteZipPath + BANG_SLASH + entryPath.replace(File.pathSeparatorChar,  '/');
```

Comment:
Shouldn't this be `File.separatorChar` instead of `File.pathSeparatorChar`?

My understanding is that `separatorChar` is the path component separator (`/` on Unix, `\` on Windows), while `pathSeparatorChar` is the `PATH` environment variable separator (`:` on Unix, `;` on Windows).

Reply:
This was fixed with the second commit.